proof/refine/ARM/KHeap_R.thy

```diff
@@ -134,6 +134,31 @@ lemma no_fail_getObject_tcb [wp]:
   apply (fastforce split: option.split_asm simp: objBits_simps' archObjSize_def)
   done
 
+lemma lookupAround2_same1[simp]:
+  "(fst (lookupAround2 x s) = Some (x, y)) = (s x = Some y)"
+  apply (rule iffI)
+   apply (simp add: lookupAround2_char1)
+  apply (simp add: lookupAround2_known1)
+  done
+
+lemma objBitsKO_bounded[simp]:
+  "objBitsKO ko < word_bits"
+  using scBits_max
+  by (simp add: objBits_simps' word_bits_def pageBits_def archObjSize_def pdeBits_def pteBits_def
+           split: Structures_H.kernel_object.split arch_kernel_object.split)
+
+lemma no_fail_getObject_sc[wp]:
```

Comment:
Don't we already have a `no_fail` rule for this or something very similar? `no_fail_getSchedContext`?